### ISSUE TITLE
[Linux] [WebRTC] rtc::scoped_refptr is deprecated:  [-Werror=deprecated-declarations]

### ISSUE THREAD

**Issue by AiraYumi** [secondlife/viewer#1455]
### Description

~~~
[78/1174] Building CXX object llwebrtc/CMakeFiles/llwebrtc.dir/llwebrtc.cpp.o
FAILED: llwebrtc/CMakeFiles/llwebrtc.dir/llwebrtc.cpp.o 
/usr/bin/c++ -DADDRESS_SIZE=64 -DAPPID=secondlife -DBOOST_BIND_GLOBAL_PLACEHOLDERS -DLL_IGNORE_SIGCHLD -D_FORTIFY_SOURCE=2 -D_REENTRANT -Dllwebrtc_EXPORTS -isystem /home/gentoo/ay_viewer/build-linux-x86_64/packages/include/webrtc -isystem /home/gentoo/ay_viewer/build-linux-x86_64/packages/include/webrtc/third_party/abseil-cpp -O2 -g -std=c++17 -fPIC -DLL_LINUX=1 -DLL_OS_DRAGDROP_ENABLED=1 -O3 -DNDEBUG -std=gnu++20 -fPIC -fexceptions -fno-math-errno -fno-strict-aliasing -fsigned-char -msse2 -mfpmath=sse -pthread -fvisibility=hidden -Werror -fno-stack-protector -Wno-parentheses -Wno-deprecated -Wno-c++20-compat -Wno-pessimizing-move -Wno-dangling-pointer -MD -MT llwebrtc/CMakeFiles/llwebrtc.dir/llwebrtc.cpp.o -MF llwebrtc/CMakeFiles/llwebrtc.dir/llwebrtc.cpp.o.d -o llwebrtc/CMakeFiles/llwebrtc.dir/llwebrtc.cpp.o -c /home/gentoo/ay_viewer/indra/llwebrtc/llwebrtc.cpp
/home/gentoo/ay_viewer/indra/llwebrtc/llwebrtc.cpp: ラムダ関数内:
/home/gentoo/ay_viewer/indra/llwebrtc/llwebrtc.cpp:214:74: エラー: ‘rtc::scoped_refptr<webrtc::AudioDeviceModule> webrtc::CreateAudioDeviceWithDataObserver(AudioDeviceModule::AudioLayer, TaskQueueFactory*, AudioDeviceDataObserver*)’ is deprecated:  [-Werror=deprecated-declarations]
  214 |             mPeerDeviceModule = webrtc::CreateAudioDeviceWithDataObserver(webrtc::AudioDeviceModule::AudioLayer::kPlatformDefaultAudio,
      |                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  215 |                                                                           mTaskQueueFactory.get(),
      |                                                                           ~~~~~~~~~~~~~~~~~~~~~~~~
  216 |                                                                           nullptr);
      |                                                                           ~~~~~~~~
次のファイルから読み込み:  /home/gentoo/ay_viewer/indra/llwebrtc/llwebrtc_impl.h:60,
         次から読み込み:  /home/gentoo/ay_viewer/indra/llwebrtc/llwebrtc.cpp:27:
/home/gentoo/ay_viewer/build-linux-x86_64/packages/include/webrtc/modules/audio_device/include/audio_device_data_observer.h:65:39: 備考: ここで宣言されています
   65 | rtc::scoped_refptr<AudioDeviceModule> CreateAudioDeviceWithDataObserver(
      |                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: すべての警告はエラーとして取り扱われます
~~~

I merged the project/webrtc-voice branch into release/maint-b and compiled it and got an error.

Here: https://github.com/AiraYumi/viewer/tree/unrelease/maint-b_with_webrtc-voice

**Comment by Nicky-D** [comment]
There was already a PR for that https://github.com/secondlife/viewer/pull/1151

Looks like it got lost in the ether

**Comment by roxanneskelly** [comment]
Oopse, got deleted when the branch was deleted due to a merge.  I'll re-apply.  Go ahead and make future PRs to project/webrtc-voice as roxie/webrtc-voice is now a 'volatile' personal development branch.

**Comment by roxanneskelly** [comment]
Okay, I've added it back to my branch, and I'll merge it into the project branch soon.

https://github.com/secondlife/viewer/commit/c7461061b8113fa258611b1a31f16a119fad1a2c

**Comment by Nicky-D** [comment]
Thank you

**Comment by AiraYumi** [comment]
Thank you.

Although I made some minor modifications, I was able to compile based on the latest code.
https://github.com/AiraYumi/viewer/commits/aira/testing/maint-b_with_webrtc-voice/

**Comment by AiraYumi** [comment]
Closes this issue as it has been fixed by "Merge Request #1532".
